detekt-core/src/main/resources/default-detekt-config.yml

```
@@ -71,6 +71,7 @@ comments:
     searchInInnerObject: true
     searchInInnerInterface: true
     searchInProtectedClass: false
+    flagCompanionWithoutName: true
```

[docs](https://kotlinlang.org/docs/object-declarations.html#companion-objects) refer to this as "unnamed companion object", so I would propose the name:
```suggestion
    ignoreUnnamedCompanionObject: false
```
which
1. matches the official terminology
2. matches params style of other rules
3. somehow makes the logic simpler (not `!`)
   _Note: normally I would advocate for a positively phrased configuration entry (`allowUnnamedCompanionObject`), but somehow in this case `ignore` makes sense as an "opt in" to more lenient behavior._

---

I agree with all of these points, I'm even a bit surprised to agree with the negative phrasing, I actually had it this way before but changed it due to positive phrasing 👍🏻